examples/vnet/doca_gw.c

```diff
@@ -17,13 +18,52 @@ struct doca_fwd_tbl {
     struct doca_fwd_table_cfg cfg;
 };
 
-struct doca_gw_pipeline {
-    void * handler;
-    uint32_t id;
-};
-
+//it's seems not need?  gw_ins->port0/port1 ??? 
 static struct doca_gw_port *doca_gw_used_ports[DOCA_GW_MAX_PORTS];
 
+static struct doca_gw_port *doca_get_port_byid(uint8_t port_id)
+{
+	return doca_gw_used_ports[port_id];
+}
+
+static struct doca_gw_port *doca_alloc_port_byid(uint8_t port_id, struct doca_gw_port_cfg *cfg)
+{
+	struct doca_gw_port *port;
+
+	port = (struct doca_gw_port *) malloc(sizeof(struct doca_gw_port) + cfg->priv_data_size);
+	if (port == NULL)
+		return NULL;
+	memset(port, 0x0, sizeof(struct doca_gw_port));
+	port->port_id = port_id;
+	doca_gw_used_ports[port_id] = port;
+	return port;
+}
+
+struct doca_gw_pipeline *doca_get_pipeline(uint16_t portid, uint8_t type)
```

Comment:
you can give a pipeline a type, for example goes to SW or hairpin, but ul/ol is application choice, so this should no be here.

Reply:
remove it . done